pkg/discovery/monitoring/kubelet/kubelet_monitor.go

```diff
@@ -126,20 +135,42 @@ func (m *KubeletMonitor) scrapeKubelet(node *api.Node) {
 		m.metricSink.AddNewMetricEntries(cacheUsedMetric)
 	}
 
-	m.parseNodeStats(summary.Node)
-	m.parsePodStats(summary.Pods)
+	currentMilliSec := time.Now().UnixNano() / int64(time.Millisecond)
+	m.parseNodeStats(summary.Node, currentMilliSec)
+	m.parsePodStats(summary.Pods, currentMilliSec)
 
 	glog.V(4).Infof("Finished scrape node %s.", node.Name)
 }
 
+// scrapeKubeletForSampling collects resource usage data samples for node, and containers and pods in the node.
+func (m *KubeletMonitor) scrapeKubeletForSampling(node *api.Node) {
```

Comment:
Suggestion: This method duplicates code, consider writing another commond method which can be used here and in `scrapeKubelet()`